src/constraint.jl

```diff
@@ -430,6 +444,147 @@ function MOI.add_constraint(m::MosekModel, xs::MOI.VectorOfVariables,
+        if 1 <= vi.value <= length(m.x_sd)
+            matrix_index = m.x_sd[vi.value]
+            i += 1
+            subi[i] = term.output_index
+            if matrix_index.matrix == 0
+                nlin += 1
+                subj[i] = 0
+                subii[i] = 0
+                subjj[i] = getindex(m.x_block, ref2id(vi))
+                cof[i]  = term.scalar_term.coefficient
+            else
+                subj[i] = matrix_index.matrix
+                subii[i] = matrix_index.row
+                subjj[i] = matrix_index.column
+                cof[i]  = term.scalar_term.coefficient
+            end
+        end
+    end
+
+    perm = Int[1:N...]
+    sort(perm,by=i->(subi[i],subj[i],subii[i],subjj[i]))
+
+    psubi  = view(subi,perm)
+    psubj  = view(subj,perm)
+    psubii = view(subii,perm)
+    psubjj = view(subjj,perm)
+    pcof   = view(cof,perm)
+
+    ptrrow = Vector{Int64}(undef,nrows)
+    nzrow  = Vector{Int32}(undef,nrows)
+    ptrrow = Vector{Int64}(undef,nrows)
+    let p = 1
+        for i in 1:nrows
+            ptrrow[i] = p
+            ptrb = p
+            ptrrow[i] = p
+            while p <= nlin && psubi[p] == i  p += 1 end
+            nzrow[i] = p-ptrb
+        end
+    end
+
+    Mosek.putafefrowlist(m.task,afeidxs,nzrow,ptrrow,psubjj[1:nlin],pcof[1:nlin])
+    Mosek.putafeglist(m.task,afeidxs,consts)
+
+    if N > nlin # psd terms
+        let p = nlin+1
+            while p <= N
+                i = psubi[p]
+                j = psubj[p]
+                ptrb = p
+                p += 1
+                while i == psubi[p] && j == psubj[p]  p += 1 end
+                midx = Mosek.appendsparsesymmat(m.task,m.sd_dim[i],psubii[ptrb:p-1],psubjj[ptrb:p-1],pcof[ptrb:p-1])
+                Mosek.putafebarfentry(m.task,i,j,[midx],[1.0])
+            end
+        end
+    end
+end
+
+function clear_afe_rows(m::MosekModel,
+                        afeidxs::Vector{Int64})
+    emptyafebarfrowlist(m.task,afeidxs)
+    N = length(afeidxs)
+    rownz = zeros(Int32,N)
+    rowptr = zeros(Int64,N)
+    putafefrowlist(m.task,afeidxs,zeros(Int32,N),zeros(Int64,N),Int32[],Float64[])
+    putafeglist(m.task,afeidxs,zeros(Float64,N))
+end
+
+if Mosek.getversion() >= (10,0,0)
+MOI.supports_constraint(mosek::MosekModel,::Type{MOI.VectorAffineFunction{Float64}},::Type{<:ACCVectorDomain}) = true
+
+function MOI.add_constraint(m::MosekModel,
+                            axb::MOI.VectorAffineFunction{Float64},
+                            dom::D) where{D<:ACCVectorDomain}
+    N = MOI.dimension(dom)
+    afeidxs = allocate_afe(m,N)
+    domidx = append_domain(m,dom)
+
+    axb = MOIU.canonical(axb)
+    set_afe_rows(m,afeidxs,axb)
+
+    accid = Mosek.getnumacc(m.task)+1
+    Mosek.appendacc(m.task,
+                    domidx,
+                    index_julia_to_mosek(afeidxs,dom),
+                    zeros(Float64,N))
+
+    ci = MOI.ConstraintIndex{MOI.VectorAffineFunction,D}(accid)
```

Comment:
Add `{Float64}` after `MOI.VectorAffiveFunction`

Reply:
Obvious when you point it out. Thanks!